ansible_ai_connect_chatbot/src/useChatbot/useChatbot.ts

```diff
@@ -0,0 +1,126 @@
+import axios from "axios";
+import { useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+import type { MessageProps } from "@patternfly/virtual-assistant/dist/dynamic/Message";
+import type {
+  ExtendedMessage,
+  ChatRequest,
+  ChatResponse,
+} from "../types/Message";
+
+export const readCookie = (name: string): string | null => {
+  const nameEQ = name + "=";
+  const ca = document.cookie.split(";");
+  for (let c of ca) {
+    const cookie = c.trim();
+    if (cookie.startsWith(nameEQ)) {
+      return cookie.substring(nameEQ.length, cookie.length);
+    }
+  }
+  return null;
+};
+
+export const botMessage = (content: string): MessageProps => {
+  return {
+    role: "bot",
+    content,
+    name: "Ansible Lightspeed Bot",
+    avatar:
+      "https://access.redhat.com/sites/default/files/images/product_icon-red_hat-ansible_automation_platform-rgb_0.png",
+    actions: {
+      // eslint-disable-next-line no-console
+      positive: { onClick: () => console.log("Good response") },
+      // eslint-disable-next-line no-console
+      negative: { onClick: () => console.log("Bad response") },
+    },
+  };
+};
+
+type AlertMessage = {
+  title: String;
+  message: String;
+  variant: "success" | "danger" | "warning" | "info" | "custom";
+};
+
+const INITIAL_NOTICE: AlertMessage = {
+  title: "Notice",
+  message: `Please do not include any personal or confidential information
+in your interaction with the virtual assistant. The tool is
+intended to assist with general queries.`,
+  variant: "info",
+};
+
+export const useChatbot = () => {
+  const [messages, setMessages] = useState<ExtendedMessage[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [alertMessage, setAlertMessage] = useState<AlertMessage | undefined>(
+    INITIAL_NOTICE,
+  );
+  const [conversationId, setConversationId] = useState<
+    string | null | undefined
+  >(undefined);
+
+  const handleSend = async (message: string) => {
+    const userMessage: ExtendedMessage = {
+      role: "user",
+      content: message,
+      name: "User",
+      avatar:
+        "https://developers.redhat.com/sites/default/files/inline-images/Skill%20development_0.png",
+      referenced_documents: [],
+    };
+    setMessages((msgs: ExtendedMessage[]) => [...msgs, userMessage]);
+
+    const chatRequest: ChatRequest = {
+      conversation_id: conversationId,
+      query: message,
+    };
+
+    setIsLoading(true);
+    try {
+      const csrfToken = readCookie("csrftoken");
+      const resp = await axios.post(
+        import.meta.env.PROD
+          ? "/api/v0/ai/chat/"
+          : "http://localhost:8080/v1/query/",
```

Comment:
@TamiTakamiya can you plz confirm why are we using `localhost` directed query API

Reply:
This is used in the debug configuration.

When `npm start` is executed, the UI runs using its own server at `http://localhost:3000`. It is handy for quick debugging, but then there is no Lightspeed srever running ang `/api/vo/ai/chat` endpoint is not available, bu if the lightspeed service (chatbot service) is running at `localhost:8080`, it can test actually chat operations.